### ISSUE TITLE
rlocus generates a *lot* of data points

### ISSUE THREAD

**Issue by murrayrm** [python-control/python-control#310]
While doing some development I noticed that running the root locus tests in the unit test suite takes a *long* time.  This occurs in `control/tests/matlab_test` and `control/tests/rlocus_tests.py`.  I did a bit of checking and it appears that the root locus command `rlocus` is generating 15K+ points in the root locus diagram, which can take a while...

Not sure when this crept in, but it would be nice for everyone (users, developers, Travis CI) to have the number of data points be a reasonable fraction of the number of pixels across a typical screen rather than a large multiple.

**Comment by murrayrm** [comment]
More details:

* For the system
```
sys1 = TransferFunction([1, 2], [1, 2, 3])
rlist, klist = rlocus(sys1)
```
the gain calculation yields 183 points (reasonable).

* For the system
```
sys2 = StateSpace([[1., 4.], [3., 2.]], [[1.], [-4.]], [[1., 0.]], [[0.]])
rlist, klist = rlocus(sys2)
```
the gain calculation yields 12549 points (unreasonable).

**Comment by murrayrm** [comment]
I chased down the problem to a calculation that is used to insert additional points in the root locus plots if the current set of points are too sparse.  There is a computation of the desired spacing on the x and y axes, and additional points are added if the initial set of points for the root locus are too far apart.  However, if the root locus plot is completely on the x axis (all roots are real), then the computation was returning a tolerance of zero and so *all* points on the root locus were being expanded by 5X => lots of new points.

I updated the calculation to look for the special case where the points are all in a line (vertical or horizontal) and then handled those as special cases.  This speeds up the root locus plot by quite a bit.

There is still some delay in setting up the zoom dispatcher.  This seems to be something associated with `matplotlib` and not our code.  I put in a TODO to look into it at some point.

PR coming soon (need to make sure the unit tests work on all platforms).
